src/WebApiContrib.Core.Versioning/Strategies/VersionContext.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace WebApiContrib.Core.Versioning
+{
+    public class VersionContext
+    {
+        public int? Version { get; }
+        public string VaryOn { get; }
+
+        public VersionContext(int? version) : this(version, null)
```

Comment:
At this point, I think we can make this required (non-nullable). You'd only return a `VersionContext` if you succeed.